src/OpenLoco/src/Objects/ObjectManager.cpp

```diff
@@ -983,8 +987,72 @@ namespace OpenLoco::ObjectManager
     // 0x004C3A9E
     void updateYearly2()
     {
-        // update available vehicles/roads/airports/etc.
-        call(0x004C3A9E);
+        const auto currentYear = getCurrentYear(); // 0x004C3AB7
+
+        // 0x004C3A9E, 004C3B8E-004C3B95
+        for (uint16_t vehicleObjId = 0; vehicleObjId < ObjectManager::getMaxObjects(ObjectType::vehicle); vehicleObjId++)
+        {
+            const auto* vehicleObject = ObjectManager::get<VehicleObject>(vehicleObjId); // 004C3AA0
+
+            if (vehicleObject == nullptr) // 004C3AA7
+            {
+                continue; // 0x004C3AAA
+            }
+
+            if (currentYear == vehicleObject->designed) // 004C3AB0-004C3ABE
+            {
+                for (Company& company : CompanyManager::companies()) // 004C3AC4-004C3AC9, 004C3B44-004C3B4F
+                {
+                    if (company.name == 0) // 004C3ACB-004C3ACF
```

Comment:
you don't need this if as `CompanyManager::companies()` iterator takes care of it skipping over unallocated companies (unallocated have 0 for company.name)